src/grammar.hh

```diff
@@ -213,6 +216,35 @@ class Grammar {
   void multi_propagate_max_filter();
 
   unsigned int to_dot(unsigned int *nodeID, std::ostream &out, int plot_level);
+
+  /* semantic check: grammar must be able to parse the empty input in
+   * order to provide recursion base for outside candidates */
+  bool check_outside_parse_empty_word();
+
+  /* input check: tests if all NTs requested by the user to be reported
+   * actually do exist in the chosen inside grammar.
+   * Throws an error if NTs are not present.
+   */
+  void check_outside_requested_nonexisting_nts();
+
+  /* not yet complete: will inject additional production rules such that
+   * the given (inside-) grammar is turned into an outside grammar.
+   */
+  void convert_to_outside();
+
+  const bool is_partof_outside() {
```

Comment:
Same thing as in previous comment.

Reply:
I went through the code changes and hopefully resolved all these situations.